File: SIRI-ISRAEL/Java/Print_sm_respone.java
package siri_sm_demo;

import java.io.FileOutputStream;
import java.io.IOException;
import java.io.OutputStreamWriter;

public class Print_sm_respone {

    public static void print_sm_respone (ServiceDeliveryStructure Response) {
        try {
            FileOutputStream fs = new FileOutputStream ("siri_sm_track.txt");
            OutputStreamWriter LogFile = new OutputStreamWriter (fs,"UTF-8");

            LogFile.write("ResponseTimestamp:         "+Response.responseTimestamp+"\n");
            LogFile.write("ProducerRef:               "+Response.producerRef.getValue()+"\n");
            LogFile.write("ResponseMessageIdentifier: "+Response.responseMessageIdentifier.getValue()+"\n");
            LogFile.write("RequestMessageRef:         "+Response.requestMessageRef.getValue()+"\n");
            LogFile.write("Status:                    "+Response.status+"\n");

            if (Response.status == false) {
                LogFile.write("There was an error\n");
                LogFile.write("ErrorText:                 "+Response.errorCondition.otherError.errorText+"\n");
                LogFile.write("Description:               "+Response.errorCondition.description.value+"\n");
            }

            if (Response.status == true) {
                int smds_num = 0;
                int msvs_num = 0;
                for (StopMonitoringDeliveryStructure smds: Response.stopMonitoringDelivery) {
                    LogFile.write("stopMonitoringDelivery num "+smds_num+" start\n");
                    LogFile.write("    ResponseTimestamp:         "+smds.responseTimestamp+"\n");
                    LogFile.write("    Status:                    "+smds.status+"\n");
                    if (smds.status == false) {
                        LogFile.write("    Error description: "+smds.errorCondition.description.getValue()+"\n");
                        LogFile.write("stopMonitoringDelivery num "+smds_num+" end\n");
                        smds_num++;
                    }
                    if (smds.status == true) {
                        msvs_num = 0;
                        for (MonitoredStopVisitStructure msvs: smds.monitoredStopVisit) {
                            LogFile.write("    monitoredStopVisit sub "+msvs_num+" start\n");
                            LogFile.write("        RecordedAtTime: "+msvs.recordedAtTime+"\n");
                            LogFile.write("        ItemIdentifier: "+msvs.itemIdentifier+"\n");
                            LogFile.write("        MonitoringRef: "+msvs.monitoringRef.getValue()+"\n");
                            LogFile.write("        LineRef: "+msvs.monitoredVehicleJourney.lineRef.getValue()+"\n");
                            LogFile.write("        DirectionRef: "+msvs.monitoredVehicleJourney.directionRef.getValue()+"\n");
                            LogFile.write("        PublishedLineName: "+msvs.monitoredVehicleJourney.publishedLineName.getValue()+"\n");
                            if (msvs.monitoredVehicleJourney.vehicleRef != null) {
                               LogFile.write("        vehicleRef: "+msvs.monitoredVehicleJourney.vehicleRef.getValue()+"\n");
                            }   
                            LogFile.write("        OperatorRef: "+msvs.monitoredVehicleJourney.operatorRef.getValue()+"\n");
                            LogFile.write("        DestinationRef: "+msvs.monitoredVehicleJourney.destinationRef.getValue()+"\n");
                            if (msvs.monitoredVehicleJourney.vehicleLocation != null) {
                               LogFile.write("        VehicleLocation: "
                                    +msvs.monitoredVehicleJourney.vehicleLocation.latitude.toString() + ","
                                    +msvs.monitoredVehicleJourney.vehicleLocation.longitude.toString()+"\n");
                            }
                            LogFile.write("        StopPointRef: "+msvs.monitoredVehicleJourney.monitoredCall.stopPointRef.getValue()+"\n");
                            LogFile.write("        OriginAimedDepartureTime: "+msvs.monitoredVehicleJourney.originAimedDepartureTime+"\n");
                            LogFile.write("        AimedArrivalTime:         "+msvs.monitoredVehicleJourney.monitoredCall.aimedArrivalTime+"\n");
                            LogFile.write("        ExpectedArrivalTime:      "+msvs.monitoredVehicleJourney.monitoredCall.expectedArrivalTime+"\n");
                        
                            LogFile.write("    monitoredStopVisit sub "+msvs_num+" end\n");
                            msvs_num++;
                        }
                        LogFile.write("stopMonitoringDelivery num "+smds_num+" end\n");
                        smds_num++;
                    }
                }
            }
            LogFile.close();
        }
        catch (IOException ex) {System.out.println("IOException:"+ex);}
    }           
    
}

File: SIRI-ISRAEL/C_sharp/Print_sm_respone.cs
using System;
using System.IO;
using Siri_sm_demo.WebReference;

namespace Siri_sm_demo
{
    class Print_sm_response {
        public static void print_sm_respone(ServiceDeliveryStructure Response) {
            try {
                int i = 65;
                StopMonitoringDeliveryStructure smds = new StopMonitoringDeliveryStructure();
                MonitoredStopVisitStructure msvs = new MonitoredStopVisitStructure();
                using (StreamWriter LogFile = new StreamWriter("siri_sm_track.txt")) {
                    LogFile.WriteLine("Start of siri_sm_track.txt");
                    LogFile.WriteLine("{0}","ResponseTimestamp:         " + Response.ResponseTimestamp);
                    LogFile.WriteLine("{0}","ProducerRef:               " + Response.ProducerRef.Value);
                    LogFile.WriteLine("{0}","ResponseMessageIdentifier: " + Response.ResponseMessageIdentifier.Value);
                    LogFile.WriteLine("{0}","RequestMessageRef:         " + Response.RequestMessageRef.Value);
                    LogFile.WriteLine("{0}","Status:                    " + Response.Status);
                    if (Response.Status == false) {
                        LogFile.WriteLine(" ");
                        LogFile.WriteLine("{0}", "There was an error");
                        LogFile.WriteLine("{0}", "ErrorText:                 " + Response.ErrorCondition.Description.Value);
                    }
                    if (Response.Status == true) {
                        for(int smds_num = 0; smds_num < Response.StopMonitoringDelivery.Length; smds_num++) {
                            LogFile.WriteLine("    Start of StopMonitoringDelivery num {0}", smds_num);
                            smds = Response.StopMonitoringDelivery[smds_num];
                            LogFile.WriteLine("{0}", "        ResponseTimestamp:         " + smds.ResponseTimestamp);
                            LogFile.WriteLine("{0}", "        Status:                    " + smds.Status);
                            if (smds.Status == false) {
                                LogFile.WriteLine("{0}", "        Error description:         " + smds.ErrorCondition.Description.Value);
                            }
                            if (smds.Status == true) {
                                for (int msvs_num = 0; msvs_num< smds.MonitoredStopVisit.Length; msvs_num++) {
                                    msvs = smds.MonitoredStopVisit[msvs_num];
                                    LogFile.WriteLine("            Start of MonitoredStopVisit num {0}", msvs_num);
                                    LogFile.WriteLine("                RecordedAtTime:           {0}",msvs.RecordedAtTime);
                                    LogFile.WriteLine("                ItemIdentifier:           {0}",msvs.ItemIdentifier);
                                    LogFile.WriteLine("                MonitoringRef:            {0}",msvs.MonitoringRef.Value);
                                    LogFile.WriteLine("                LineRef:                  {0}",msvs.MonitoredVehicleJourney.LineRef.Value);
                                    LogFile.WriteLine("                DirectionRef:             {0}",msvs.MonitoredVehicleJourney.DirectionRef.Value);
                                    LogFile.WriteLine("                PublishedLineName:        {0}",msvs.MonitoredVehicleJourney.PublishedLineName.Value);
                                    if (msvs.MonitoredVehicleJourney.VehicleRef == null) {
                                        LogFile.WriteLine("                vehicleRef:               is null");
                                    }
                                    else {
                                        LogFile.WriteLine("                vehicleRef:               {0}", msvs.MonitoredVehicleJourney.VehicleRef.Value);
                                    }
                                    LogFile.WriteLine("                OperatorRef:              {0}",msvs.MonitoredVehicleJourney.OperatorRef.Value);
                                    LogFile.WriteLine("                DestinationRef:           {0}",msvs.MonitoredVehicleJourney.DestinationRef.Value);
                                    if (msvs.MonitoredVehicleJourney.VehicleLocation == null) {
                                        LogFile.WriteLine("                VehicleLocation:          is null");
                                    }
                                    else { 
                                        LogFile.WriteLine("                VehicleLocation:          {0}",
                                               msvs.MonitoredVehicleJourney.VehicleLocation.Items[0] + ","
                                             + msvs.MonitoredVehicleJourney.VehicleLocation.Items[1]);
                                    }

                                    LogFile.WriteLine("                StopPointRef:             {0}", msvs.MonitoredVehicleJourney.MonitoredCall.StopPointRef.Value);
                                    LogFile.WriteLine("                OriginAimedDepartureTime: {0}", msvs.MonitoredVehicleJourney.OriginAimedDepartureTime);
                                    LogFile.WriteLine("                AimedArrivalTime:         {0}", msvs.MonitoredVehicleJourney.MonitoredCall.AimedArrivalTime);
                                    LogFile.WriteLine("                ExpectedArrivalTime:      {0}", msvs.MonitoredVehicleJourney.MonitoredCall.ExpectedArrivalTime);
                                    LogFile.WriteLine("            End   of MonitoredStopVisit num {0}", msvs_num);
                                }

                            }
                            LogFile.WriteLine("    End of StopMonitoringDelivery num {0}", smds_num);
                        }
                    }
                    LogFile.WriteLine("End of siri_sm_track.txt");
                }
            }
            catch (Exception e) {Console.WriteLine("{0} Exception caught.", e);}
        }
    }
}